Aligned some features closer to native Ring behavior

SEKCobra opened the pull request:
Made it so you can directly switch between armed modes without having to disarm first. Also extended it so you could change mode while it's still arming, although this is not currently supported by Alarmo (It simply won't process the new arm service call). Also adjusted the countdown event so it uses the next highest countdown feedback, I think this is better than hardcoding the exact amounts, because the final "armed" or "alarm" event will interrupt it anyway, so even though the countdown doesn't quite get to finish if for example set to 10 seconds and using the 15 second countdown, the behavior is still sensible and better than the fallback error beep. Added panic alarm handling, right now this only sets the alarm to pending, but I am already in touch with the Alarmo devs to get that corrected.

SEKCobra committed:
Aligned some features closer to native Ring behavior

Made it so you can directly switch between armed modes without having to disarm first. Also extended it so you could change mode while it's still arming, although this is not currently supported by Alarmo (It simply won't process the new arm service call).
Also adjusted the countdown event so it uses the next highest countdown feedback, I think this is better than hardcoding the exact amounts, because the final "armed" or "alarm" event will interrupt it anyway, so even though the countdown doesn't quite get to finish if for example set to 10 seconds and using the 15 second countdown, the behavior is still sensible and better than the fallback error beep.
Added panic alarm handling, right now this only sets the alarm to pending, but I am already in touch with the Alarmo devs to get that corrected.

Dinth commented:
Hi @SEKCobra, have you managed to get those couple of changes merged with the upstream Alarmo?

SEKCobra commented:
> Hi @SEKCobra, have you managed to get those couple of changes merged with the upstream Alarmo?

Unfortunately it seems my patch was never accepted.

Dinth commented:
Would you mind putting a link to the PR here for visibility? Maybe we can show some support for the changes on their GH

SEKCobra commented:
> Would you mind putting a link to the PR here for visibility? Maybe we can show some support for the changes on their GH

Sorry, I was getting confused myself about the state of things. Alarmo is updated and supports the behavior fully, probably close to two years.
The Home Assistant Template was updated with the changes referenced here, but I believe I made some further improvements that have not yet been integrated, I am discovering right now that I might have forgotten to add a pull request for that. I'll have to check my current version and see if anything needs to be merged.

SEKCobra committed:
Update keypad-blueprint-v1.yaml

Removed notes about not being supported by Alarmo, as the feature has been implemented by Alarmo.

SEKCobra committed:
Merge branch 'ImSorryButWho:main' into patch-2

SEKCobra commented:
Ok, got my bearings back, this is the open pull request implementing the fixes I have made. Unfortunately this pull request is still open and therefore not integrated upstream.

palves commented:
I'm going to hazzard a guess that one reason this hasn't been merged is that the PR does too many things at once.  If you split it to one PR per feature (and a clean rebased PR without merges and fixes to fixes), it'll probably be much easier to review and accept.
